pystac/collection.py

```diff
@@ -76,6 +76,30 @@ def __init__(self,
         self.properties = properties
         self.summaries = summaries
 
+    def set_self_href(self, href):
+        """Sets the absolute HREF that is represented by the ``rel == 'self'``
+        :class:`~pystac.Link`.
+
+        Args:
+            str: The absolute HREF of this object. If the given HREF
+                is not absolute, it will be transformed to an absolute
+                HREF based on the current working directory.
+
+        Note:
+            Overridden for collections so that the root's ResolutionObjectCache can properly
+            update the HREF cache.
+        """
+        root = self.get_root()
+        if root is not None:
+            root._resolved_objects.remove(self)
+
+        super().set_self_href(href)
+
+        if root is not None:
+            root._resolved_objects.cache(self)
```

Comment:
I just want to make sure this is making sense here because at first pass it seemed redundant to me: un-resolve the collection to just re-resolve it. Is the idea that in line 96, the href is set correctly according to the Catalog_Type of root?

Reply:
Yeah, removing it from the cache de-registers it with the previous HREF - collections are cached by href in order to facilitate the collection metadata merging (the thing that moved to the commons extension in 1.0). The confusing logic is a good point and I'll add comments to clarify.

Reply:
Ok, that makes sense